### PR TITLE
Remove dependency on deprecated github.com/pkg/errors

### PR DESCRIPTION
**Pull request by imjasonh**
This package is deprecated, archived, and in maintenance mode, since Go
errors support wrapping natively.

Signed-off-by: Jason Hall <jason@chainguard.dev>

Progress toward https://github.com/open-policy-agent/opa/issues/2152

### CODE REVIEW

File: ast/location/location.go [hunk 1]
```diff
@@ -36,12 +36,6 @@ func (loc *Location) Errorf(f string, a ...interface{}) error {
 	return errors.New(loc.Format(f, a...))
 }
 
-// Wrapf returns a new error value that wraps an existing error with a message formatted
-// to include the location info (e.g., line, column, filename, etc.)
-func (loc *Location) Wrapf(err error, f string, a ...interface{}) error {
```
Comment (srenatus) [inline ast/location/location.go:41]: Could we do `return fmt.Errorf(f+": %w", append(a, err))` instead? 🤔 We wouldn't have to remove the exported method then, I believe...

**Review by srenatus — COMMENTED** [review]
Thanks for picking this up! 🎉 One quick remark, I'll give this a more thorough review tomorrow.

File: ast/location/location.go (hunk 1 above)
Comment (imjasonh) [reply ast/location/location.go:41]: Sure, I wasn't sure if this was actually used or necessary, but I suppose it's better not to drop an exported method if you can avoid it.

**Comment by imjasonh** [comment]
The lint failure in https://github.com/open-policy-agent/opa/runs/6458114145?check_suite_focus=true actually demonstrates another reason we should prefer stdlib's wrapped errors -- linter checks don't know that `errors.Wrapf(...)` is an error with a format string, so they can't fail when the format string parts don't align with the passed values. `fmt.Errorf` carries that semantic meaning, so linters can check for it. 👍

File: internal/jwx/jws/verify/hmac.go [hunk 2]
```diff
@@ -23,11 +22,11 @@ func (v HMACVerifier) Verify(signingInput, signature []byte, key interface{}) (e
 
 	expected, err := v.signer.Sign(signingInput, key)
 	if err != nil {
-		return errors.Wrap(err, `failed to generated signature`)
+		return fmt.Errorf("failed to generated signature: %w", err)
 	}
 
 	if !hmac.Equal(signature, expected) {
-		return errors.New(`failed to match hmac signature`)
+		return fmt.Errorf("failed to match hmac signature: %w", err)
```
Comment (zchee) [inline internal/jwx/jws/verify/hmac.go:29]:
```suggestion
		return errors.New("failed to match hmac signature")
```

**Review by zchee — CHANGES_REQUESTED** [review]
cc: @imjasonh 

Fix CI error with below diff:

```diff
diff --git a/bundle/verify_test.go b/bundle/verify_test.go
index 4021b498..e82f2249 100644
--- a/bundle/verify_test.go
+++ b/bundle/verify_test.go
@@ -178,7 +178,7 @@ yQjtQ8mbDOsiLLvh7wIDAQAB==
 		"public_key_with_invalid_HS256_sign": {
 			signedTokenHS256,
 			map[string]*KeyConfig{"foo": {Key: "bad_secret", Algorithm: "HS256"}}, "", "",
-			true, fmt.Errorf("Failed to verify message: failed to match hmac signature"),
+			true, fmt.Errorf("failed to verify message: failed to match hmac signature"),
 		},
 		"public_key_with_valid_RS256_sign": {
 			signedTokenRS256,
@@ -188,7 +188,7 @@ yQjtQ8mbDOsiLLvh7wIDAQAB==
 		"public_key_with_invalid_RS256_sign": {
 			signedTokenRS256,
 			map[string]*KeyConfig{"foo": {Key: publicKeyInvalid, Algorithm: "RS256"}}, "", "",
-			true, fmt.Errorf("Failed to verify message: crypto/rsa: verification error"),
+			true, fmt.Errorf("failed to verify message: crypto/rsa: verification error"),
 		},
 		"public_key_with_bad_cert_RS256": {
 			signedTokenRS256,
```

**Comment by srenatus** [comment]
image

Interesting. You're the first person I've come to meet using Vigilant mode. Would you think this is something to fix? I guess you could also sign the one commit we'll eventually squash this into... OTOH I don't know what exactly would happen to that when I press "squash and merge" anyways... 🤔

**Comment by imjasonh** [comment]
> image
> 
> Interesting. You're the first person I've come to meet using Vigilant mode. Would you think this is something to fix? I guess you could also sign the one commit we'll eventually squash this into... OTOH I don't know what exactly would happen to that when I press "squash and merge" anyways... 🤔

Nice of you to notice! 😆  I'm using [`gitsign`](https://github.com/sigstore/gitsign) to sign commits using an ephemeral cert from Sigstore's root CA, Fulcio. The issue is that GitHub doesn't know to trust the CA, so the signature shows up as "Unverified" in the UI for now. 😞 

If this is any kind of problem whatsoever for this project I'm happy to squash and remove the signatures. I also don't mind if you squash and commit and the signatures get dropped in the process. I'm mostly just dogfooding this relatively new tool.

**Comment by srenatus** [comment]
> If this is any kind of problem whatsoever for this project I'm happy to squash and remove the signatures. I also don't mind if you squash and commit and the signatures get dropped in the process. I'm mostly just dogfooding this relatively new tool.

Thanks for the insights. It's fine either way here. I suppose we'll go with squash and merge and it'll be dropped; if it's unverified in the meantime, it's now clear what's going on. Thanks again!

**Review by srenatus — APPROVED** [review]

**Comment by srenatus** [comment]
Thanks for also making lower-case beginnings of the error strings more consistent! 👍

**Pull request merged**